### ISSUE TITLE
Panic with SIGSEGV from influxd v2 alpha

### ISSUE THREAD

**Issue by tmalaher** [influxdata/influxdb#16131]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Downloaded Linux binaries on Dec 4, 2019, running on:
```
$ uname -a
Linux btln007674 3.10.0-1062.7.1.el7.x86_64 #1 SMP Tue Nov 26 09:08:45 PST 2019 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/oracle-release
Oracle Linux Server release 7.7
```
2. Used the web-based setup to get started
3. Configured telegraf to push system info to influxd
4. Created some command line `ingest write` scripts to populate some more data
5. At some point, while trying queries and setting up some dashboards:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x91daf3]

goroutine 115694 [running]:
go.uber.org/zap.(*Logger).check(0x0, 0x1ffffffffffff00, 0x1c2bbb0, 0x29, 0x44bcd7)
        /go/pkg/mod/go.uber.org/zap@v1.9.1/logger.go:264 +0x93
go.uber.org/zap.(*Logger).Info(0x0, 0x1c2bbb0, 0x29, 0x0, 0x0, 0x0)
        /go/pkg/mod/go.uber.org/zap@v1.9.1/logger.go:186 +0x44
github.com/influxdata/influxdb/nats.(*AsyncPublisher).Publish.func1(0xc057b82960, 0x16, 0x28f3ec0, 0xc0003ef220)
        /go/src/github.com/influxdata/influxdb/nats/publisher.go:44 +0x6c
github.com/nats-io/go-nats-streaming.(*conn).cleanupOnClose(0xc00004e1c0, 0x28f3ec0, 0xc0003ef220)
        /go/pkg/mod/github.com/nats-io/go-nats-streaming@v0.4.0/stan.go:506 +0x27a
github.com/nats-io/go-nats-streaming.(*conn).closeDueToPing(0xc00004e1c0, 0x28f3ec0, 0xc0003ef220)
        /go/pkg/mod/github.com/nats-io/go-nats-streaming@v0.4.0/stan.go:462 +0x61
github.com/nats-io/go-nats-streaming.(*conn).pingServer(0xc00004e1c0)
        /go/pkg/mod/github.com/nats-io/go-nats-streaming@v0.4.0/stan.go:414 +0x1c3
created by time.goFunc
        /usr/local/go/src/time/sleep.go:169 +0x44
```

After a restart:

```sh
curl -o vars.txt "http://localhost:8086/debug/vars"
```

```json
{
"cmdline": ["./influxd","--http-bind-address",":1844","--bolt-path","/emit/influx/influxdbv2/influxd.bolt","--engine-path","/emit/influx/influxdbv2/engine","--reporting-disabled"],
"memstats": {"Alloc":104779192,"TotalAlloc":5403349608,"Sys":143952120,"Lookups":0,"Mallocs":78700882,"Frees":77560402,"HeapAlloc":104779192,"HeapSys":131760128,"HeapIdle":24281088,"HeapInuse":107479040,"HeapReleased":0,"HeapObjects":1140480,"StackInuse":2457600,"StackSys":2457600,"MSpanInuse":1202832,"MSpanSys":1327104,"MCacheInuse":6944,"MCacheSys":16384,"BuckHashSys":2067969,"GCSys":5464064,"OtherSys":858871,"NextGC":113543376,"LastGC":1575502270127738144,"PauseTotalNs":102741903,"PauseNs":[26089,30215,23904,28987,22170,41860,37376,25514,25114,24305,70275,52230,31617,80839,23086,30444,29176,31829,24891,29630,42437,24389,22727,22774,44554,21493,25500,29964,24564,30003,24110,42729,20934,122990,60278,40199,30100,20001,64841,26487,111153,85244,21461,78693,29836,22263,23497,25523,43920,28816,28570,28110,27847,34662,34003,25741,73952,100416,25886,28147,32076,23289,32087,24512,28626,52703,25031,23931,28040,170510,218016,256919,213002,203403,200505,201838,204224,205637,196379,221659,266113,228991,94609,695610,34551,212370,249222,218540,251268,326358,237016,262278,250045,241080,236082,59860,281696,224694,193172,103655,214229,220873,219589,213306,237265,92996,236488,217419,48201,221884,80571,15054,17110,51177,196090,388351,58930,36344,200319,207587,33565,194124,14754,414926,230801,386990,292406,231955,213236,207231,240540,253987,213088,208051,215739,234989,17046,20039,58391,39746,26372,28687,43579,36412,28355,96660,72229,38880,28980,22058,20891,38817,36138,18610,23409,25737,396315,20349,84732,24210,24628,60888,103752,49441,38521,34212,113855,29918,24282,23122,23094,24797,25403,33208,29733,19771,20589,26593,37597,23012,25779,27619,23934,24903,29693,50734,40622,25151,27670,112804,25098,23599,33086,26599,17244,23760,22282,235657,33072,71160,36826,22895,32498,99053,59458,112307,110979,90726,49138,262320,207781,56431,43087,49859,36969,32272,38580,54067,31066,32245,64988,23252,218529,33048,17559,21816,43912,26073,19208,24044,1309591,28983,35888,24566,19715,95094,32483,24188,29290,23568,32543,20248,43157,60302,31007,21344,21669,72168,27226,27912,24502,21838,4430443,59413,55553,31817],"PauseEnd":[1575500649679617932,1575500669659476694,1575500679696766284,1575500699674902962,1575500719658188495,1575500730030814519,1575500749662520974,1575500769651284257,1575500779687039894,1575500799680636574,1575500819670899961,1575500840031771721,1575500860035671945,1575500880041575787,1575500890082802376,1575500910056813183,1575500930056530194,1575500950061337494,1575500970061288625,1575500990063671665,1575501020012188192,1575501038875483573,1575501050030780128,1575501070037970949,1575501090043222080,1575501110050207413,1575501130081197765,1575501160056891704,1575501180059435336,1575501200067640304,1575501230037884606,1575501250064880566,1575501270085037394,1575501300058624222,1575501320071391662,1575501350066660703,1575501370095434736,1575501400071660209,1575501428824228945,1575501450046141133,1575501470093682903,1575501500062202666,1575501520092842476,1575501550103157890,1575501580050440821,1575501600413173963,1575501630080122213,1575501660091836037,1575501683972515933,1575501687015226323,1575501710034173939,1575501730087553163,1575501760060718864,1575501780397684782,1575501800099169044,1575501806386535229,1575501820094946032,1575501850100286336,1575501880131237027,1575501920062237273,1575501950096603181,1575501980090577572,1575502020074580348,1575502050106853679,1575502090087457864,1575502120104210795,1575502160090050485,1575502200077103769,1575502230126309144,1575502270127738144,1575499888154260981,1575499888161748141,1575499888169457217,1575499888177167878,1575499888184996251,1575499888193911334,1575499888202186351,1575499888210372954,1575499888217713169,1575499888225493935,1575499888233871464,1575499888241154690,1575499888248552514,1575499888256503142,1575499888266710249,1575499888276652629,1575499888289943380,1575499888305155074,1575499888319682666,1575499888338101062,1575499888352938187,1575499888362156798,1575499888373773355,1575499888384850856,1575499888396646300,1575499888408598943,1575499888422643163,1575499888433301914,1575499888441036357,1575499888449739005,1575499888457458957,1575499888465579541,1575499888472915738,1575499888480847576,1575499888488720507,1575499888497371254,1575499888505716813,1575499888514112904,1575499888522180901,1575499888531751732,1575499888539759090,1575499888548461520,1575499888560354600,1575499888573216088,1575499888582354824,1575499888593259635,1575499888602814554,1575499888611917394,1575499888621561640,1575499888629930412,1575499888637827130,1575499888647459254,1575499888656510853,1575499888673862398,1575499888697104040,1575499888713815145,1575499888722300143,1575499888734434494,1575499888742599634,1575499888750984818,1575499888766023051,1575499888779091849,1575499888801419842,1575499888823588695,1575499888833008103,1575499888842781054,1575499888853720481,1575499888863750236,1575499888877504927,1575499888897468947,1575499888911366726,1575499888931700392,1575499888956821754,1575499888984755112,1575499889015313138,1575499889045355618,1575499889074171852,1575499889094601161,1575499889119506461,1575499889147114517,1575499889175428583,1575499889201779878,1575499889225999433,1575499889252300076,1575499889279771600,1575499889308228598,1575499897695880282,1575499897725443407,1575499900217237427,1575499910716032248,1575499916807123826,1575499917829938818,1575499919883700762,1575499922108075103,1575499926561400608,1575499926662575914,1575499926936386915,1575499927045304284,1575499929851469454,1575499929942215308,1575499930081727593,1575499931537531842,1575499931689634884,1575499933550244516,1575499933643615225,1575499946931276960,1575499955116145865,1575499955212765823,1575499955329403936,1575499956797512353,1575499959989625788,1575499960078381366,1575499960420340731,1575499960987992236,1575499963663904290,1575500002967809218,1575500015630212653,1575500015665307063,1575500015723670104,1575500015761485140,1575500015807405964,1575500045631171191,1575500049085827655,1575500049167897611,1575500049549837816,1575500053002395446,1575500054045619936,1575500056814862008,1575500176912090129,1575500239652831377,1575500249650642267,1575500267030258506,1575500272012040487,1575500272208921575,1575500272231368243,1575500272270217870,1575500272295932347,1575500272319180724,1575500272348461489,1575500272374265236,1575500272400357034,1575500272443711910,1575500272482299548,1575500279646182276,1575500284814644741,1575500284841963509,1575500284876894560,1575500284902523338,1575500289673741244,1575500299674612615,1575500310024548264,1575500319686304596,1575500339626072558,1575500349649359130,1575500359653404931,1575500369673893066,1575500379684725841,1575500399637415542,1575500409654117541,1575500419668993422,1575500430023176066,1575500449628536422,1575500459660055531,1575500469667798213,1575500489626423040,1575500499659170429,1575500509683798116,1575500529651156059,1575500539684366765,1575500559642586969,1575500562368518405,1575500562543904254,1575500562768935039,1575500567729039251,1575500567881843348,1575500568479393885,1575500568666702632,1575500570035384576,1575500570864566557,1575500571292401334,1575500579675377778,1575500589655758789,1575500600030984898,1575500609646260040,1575500619653815447,1575500629702982903],"NumGC":582,"NumForcedGC":0,"GCCPUFraction":0.0004883759629076571,"EnableGC":true,"DebugGC":false,"BySize":[{"Size":0,"Mallocs":0,"Frees":0},{"Size":8,"Mallocs":695079,"Frees":681936},{"Size":16,"Mallocs":33501920,"Frees":32674558},{"Size":32,"Mallocs":5033903,"Frees":4977933},{"Size":48,"Mallocs":8426990,"Frees":8346625},{"Size":64,"Mallocs":4798702,"Frees":4776989},{"Size":80,"Mallocs":678858,"Frees":669717},{"Size":96,"Mallocs":950569,"Frees":937794},{"Size":112,"Mallocs":798779,"Frees":786748},{"Size":128,"Mallocs":2408113,"Frees":2367673},{"Size":144,"Mallocs":214082,"Frees":211700},{"Size":160,"Mallocs":207460,"Frees":204459},{"Size":176,"Mallocs":54844,"Frees":54337},{"Size":192,"Mallocs":111104,"Frees":110254},{"Size":208,"Mallocs":95202,"Frees":94433},{"Size":224,"Mallocs":58171,"Frees":57741},{"Size":240,"Mallocs":757890,"Frees":745193},{"Size":256,"Mallocs":71578,"Frees":70832},{"Size":288,"Mallocs":4542696,"Frees":4517930},{"Size":320,"Mallocs":349724,"Frees":341426},{"Size":352,"Mallocs":77357,"Frees":76619},{"Size":384,"Mallocs":287535,"Frees":282178},{"Size":416,"Mallocs":32584,"Frees":32536},{"Size":448,"Mallocs":25288,"Frees":25249},{"Size":480,"Mallocs":4102,"Frees":4087},{"Size":512,"Mallocs":17915,"Frees":17694},{"Size":576,"Mallocs":78927,"Frees":77622},{"Size":640,"Mallocs":1544,"Frees":1468},{"Size":704,"Mallocs":1362,"Frees":1334},{"Size":768,"Mallocs":12064,"Frees":11909},{"Size":896,"Mallocs":3330,"Frees":3254},{"Size":1024,"Mallocs":9971,"Frees":9638},{"Size":1152,"Mallocs":47656,"Frees":46843},{"Size":1280,"Mallocs":6630,"Frees":6564},{"Size":1408,"Mallocs":3342,"Frees":3293},{"Size":1536,"Mallocs":13813,"Frees":13638},{"Size":1792,"Mallocs":4551,"Frees":4509},{"Size":2048,"Mallocs":6485,"Frees":6235},{"Size":2304,"Mallocs":12281,"Frees":12127},{"Size":2688,"Mallocs":2213,"Frees":2180},{"Size":3072,"Mallocs":4755,"Frees":4679},{"Size":3200,"Mallocs":14,"Frees":12},{"Size":3456,"Mallocs":47,"Frees":47},{"Size":4096,"Mallocs":6887,"Frees":5293},{"Size":4864,"Mallocs":5042,"Frees":4980},{"Size":5376,"Mallocs":971,"Frees":943},{"Size":6144,"Mallocs":4965,"Frees":4872},{"Size":6528,"Mallocs":31,"Frees":27},{"Size":6784,"Mallocs":22,"Frees":19},{"Size":6912,"Mallocs":7,"Frees":7},{"Size":8192,"Mallocs":4560,"Frees":3832},{"Size":9472,"Mallocs":2379,"Frees":2344},{"Size":9728,"Mallocs":113,"Frees":110},{"Size":10240,"Mallocs":120,"Frees":118},{"Size":10880,"Mallocs":87,"Frees":86},{"Size":12288,"Mallocs":1401,"Frees":1370},{"Size":13568,"Mallocs":272,"Frees":269},{"Size":14336,"Mallocs":98,"Frees":95},{"Size":16384,"Mallocs":2708,"Frees":2666},{"Size":18432,"Mallocs":193,"Frees":185},{"Size":19072,"Mallocs":45,"Frees":44}]}
}

```

**Comment by russorat** [comment]
@tmalaher thanks for reporting! We will take a look.

**Comment by jacobmarble** [comment]
I'm not familiar with this code, but.

Looks like `AsyncPublisher.Logger` is nil. A quick scan of the code confirms this. The reason we aren't seeing the panic frequently is that we only use the logger when NATS publish returns an error, which is uncommon.

I fixed the nil pointer in #15996, so the bug is fixed in master, and will be part of the next (alpha-20) release.
